cstrml/swork/src/lib.rs

```diff
@@ -747,23 +742,38 @@ impl<T: Config> Module<T> {
     /// 1. passive check work report: judge if the work report is outdated
     /// 2. (maybe) set corresponding storage order to failed if wr is outdated
     /// 2. return the (reserved, used) storage of this reporter account
-    fn get_workload(anchor: &SworkerAnchor, current_rs: u64) -> (u128, u128) {
+    fn get_workload(reporter: &T::AccountId, id: &Identity<T::AccountId>, current_rs: u64) -> (u128, u128) {
         // Got work report
-        if let Some(wr) = Self::work_reports(anchor) {
-            if Self::reported_in_slot(anchor, current_rs) {
+        if let Some(wr) = Self::work_reports(&id.anchor) {
+            if Self::can_add_stake_limit(reporter, id, current_rs) {
                 return (wr.free as u128, wr.used as u128)
             }
         }
         // Or nope, idk wtf? 🙂
         log!(
             debug,
             "🔒 No workload for anchor {:?} in slot {:?}",
-            anchor,
+            &id.anchor,
             current_rs
         );
         (0, 0)
     }
 
+    fn can_add_stake_limit(reporter: &T::AccountId, id: &Identity<T::AccountId>, current_rs: u64) -> bool {
```

Comment:
Change to *is_fully_reported*? Cause there is no concept of stake limit in MPoW protocol